src/luascript.cpp

```diff
@@ -8097,8 +8097,8 @@ int LuaScriptInterface::luaPlayerSetGuildLevel(lua_State* L)
 	// player:setGuildLevel(level)
 	uint8_t level = getNumber<uint8_t>(L, 2);
 	Player* player = getUserdata<Player>(L, 1);
-	if (player) {
-		player->setGuildLevel(level);
+	if (player && player->getGuild()) {
+		player->setGuildRank(player->getGuild()->getRankByLevel(level));
```

Comment:
You should push false to Lua if getRankByLevel returns null, and not set the guild rank in that case. It could cause a null pointer dereference in Player.getGuildLevel() if the player is in a guild but the rank is null.